src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java19/InputJava19PatternsInSwitchLabels.java

```diff
@@ -0,0 +1,71 @@
+//non-compiled with javac: Compilable with Java19
+package com.puppycrawl.tools.checkstyle.grammar.java19;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+public class InputJava19PatternsInSwitchLabels {
+    static void m1(Object o) {
+        // patterns in switch are guarded by 'when'
+        switch (o) {
+            case Integer i when i >= 0 -> System.out.println(i);
+            case Integer i -> System.out.println(i << 1);
+            default -> throw new IllegalStateException(
+                    "Unexpected value: " + o);
+        }
+    }
+
+    static <T> void m2(Object o) {
+        // Note that pattern variable cannot be cast to generic type
+        switch (o) {
+            case List list1 when (list1.size() >= 0
+                    && list1.size() < 10) || list1.equals(new ArrayList<>()) ->
+                    System.out.println(list1);
+            case List list2 when (list2.size() >= 0
+                    && list2.size() < 20) || list2.equals(new LinkedList()) ->
+                    System.out.println("list2");
+            case List list3 when list3.size() >= 0
+                    && list3.size() < 45 || list3.equals(new LinkedList()) ->
+                    System.out.println("list3");
+            case Set set when "test".equals(set.toString()) ->
+                    System.out.println(set);
+            default -> throw new IllegalStateException(
+                    "Unexpected value: " + o);
+        }
+    }
+
+    static String m3(Object o) {
+        // patterns in switch are guarded by 'when'
+        return switch (o) {
+            case List list1 when list1.size() >= 0
```

Comment:
too much obsession with only `>=` in simple examples.

use some/all others in other examples.

Reply:
Created https://github.com/checkstyle/checkstyle/pull/12531/files#diff-bd5e6e07cacf7956c943d4b551aac1b4875f6697ea16304bd9ef4a2d5fe4d086, all operators in the Java language are covered in most simple switch label possible.